File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/java/io/github/lishangbu/avalon/oauth2/common/core/AuthorizationGrantTypeSupportTest.java
package io.github.lishangbu.avalon.oauth2.common.core;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
import org.springframework.security.oauth2.core.AuthorizationGrantType;

class AuthorizationGrantTypeSupportTest {

    @Test
    void passwordGrantTypeMatchesStandardValue() {
        AuthorizationGrantType expected = new AuthorizationGrantType("password");

        assertEquals(expected, AuthorizationGrantTypeSupport.PASSWORD);
        assertEquals("password", AuthorizationGrantTypeSupport.PASSWORD.getValue());
    }

    @Test
    void smsGrantTypeMatchesCustomValue() {
        AuthorizationGrantType expected = new AuthorizationGrantType("sms");

        assertEquals(expected, AuthorizationGrantTypeSupport.SMS);
        assertEquals("sms", AuthorizationGrantTypeSupport.SMS.getValue());
    }

    @Test
    void emailGrantTypeMatchesCustomValue() {
        AuthorizationGrantType expected = new AuthorizationGrantType("email");

        assertEquals(expected, AuthorizationGrantTypeSupport.EMAIL);
        assertEquals("email", AuthorizationGrantTypeSupport.EMAIL.getValue());
    }
}

File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/test/kotlin/io/github/lishangbu/avalon/oauth2/common/core/AuthorizationGrantTypeSupportTest.kt
package io.github.lishangbu.avalon.oauth2.common.core

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.security.oauth2.core.AuthorizationGrantType

class AuthorizationGrantTypeSupportTest {
    @Test
    fun passwordGrantTypeMatchesStandardValue() {
        val expected: AuthorizationGrantType = AuthorizationGrantType("password")

        assertEquals(expected, AuthorizationGrantTypeSupport.PASSWORD)
        assertEquals("password", AuthorizationGrantTypeSupport.PASSWORD.getValue())
    }

    @Test
    fun smsGrantTypeMatchesCustomValue() {
        val expected: AuthorizationGrantType = AuthorizationGrantType("sms")

        assertEquals(expected, AuthorizationGrantTypeSupport.SMS)
        assertEquals("sms", AuthorizationGrantTypeSupport.SMS.getValue())
    }

    @Test
    fun emailGrantTypeMatchesCustomValue() {
        val expected: AuthorizationGrantType = AuthorizationGrantType("email")

        assertEquals(expected, AuthorizationGrantTypeSupport.EMAIL)
        assertEquals("email", AuthorizationGrantTypeSupport.EMAIL.getValue())
    }
}
